packages/ui/src/views/agentflows/index.jsx

```diff
@@ -58,6 +59,18 @@ const Agentflows = () => {
     const getAllAgentflowsApi = useApi(chatflowsApi.getAllAgentflows)
     const getMarketplaceAgentflowsApi = useApi(marketplacesApi.getAllTemplatesFromMarketplaces)
 
+    const renderItemCard = ({ item, images, nodeTypes, onClick, type, updateFlowsApi, setError }) => (
```

Comment:
This function is defined on every render, will produce unnecessary re-renders and can hurt performance heavily.